### PR TITLE
[Merged by Bors] - feat(Order/Interval/Set/Monotone): strictMono_of_lt_succ

### PR DESCRIPTION
**Pull request by YnirPaz**
Prove `strictMono_of_lt_succ` and `strictAnti_of_pred_lt` for archimedean orders.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by vihdzp** [comment]
Since your initial PR on club sets, I made #16041 which adds the function `toNat : Ordinal → ℕ`. I just added `omegaIso` to that PR, defined in terms of `toNat` instead.

**Comment by YaelDillies** [comment]
Is `strictMonoOn_Iic_of_lt_succ` exactly what you're looking for? I understand what's wrong with it. Or do you want a version for `Iio` instead? Should be easy to prove from there.

**Comment by YnirPaz** [comment]
> Is `strictMonoOn_Iic_of_lt_succ` not exactly what you're looking for? I understand what's wrong with it. Or do you want a version for `Iio` instead? Should be easy to prove from there.

If #15950 will give us `IsSuccArchimedean (Iio ω)` then we just need `strictMono_of_lt_succ` with the preorder assumption replaced with a linear order, right? It's trivial from `strictMonoOn_Iic_of_lt_succ` but doesn't seem to exist. I can just prove it here and then `strictMono_of_succ_lt_omega` follows from `IsSuccArchimedean (Iio ω)`.

File: Mathlib/Order/Interval/Set/Monotone.lean [hunk 1]
```diff
@@ -188,6 +188,10 @@ theorem strictMonoOn_Iic_of_lt_succ [SuccOrder α] [IsSuccArchimedean α] {n : 
   refine hψ _ (lt_of_lt_of_le ?_ hy)
   rwa [Function.iterate_succ', Function.comp_apply, lt_succ_iff_not_isMax]
 
+theorem strictMono_of_lt_succ [SuccOrder α] [IsSuccArchimedean α]
+    (hψ : ∀ m, ψ m < ψ (succ m)) : StrictMono ψ := fun _ _ h ↦
+  (strictMonoOn_Iic_of_lt_succ fun m _ ↦ hψ m) h.le (le_refl _) h
```
Comment (YaelDillies) [inline Mathlib/Order/Interval/Set/Monotone.lean:193]:
```suggestion
  (strictMonoOn_Iic_of_lt_succ fun m _ ↦ hψ m) h.le le_rfl h
```

File: Mathlib/Order/Interval/Set/Monotone.lean [hunk 2]
```diff
@@ -188,6 +188,10 @@ theorem strictMonoOn_Iic_of_lt_succ [SuccOrder α] [IsSuccArchimedean α] {n : 
   refine hψ _ (lt_of_lt_of_le ?_ hy)
   rwa [Function.iterate_succ', Function.comp_apply, lt_succ_iff_not_isMax]
 
+theorem strictMono_of_lt_succ [SuccOrder α] [IsSuccArchimedean α]
```
Comment (YaelDillies) [inline Mathlib/Order/Interval/Set/Monotone.lean:191]: Can you also add `strictAnti_of_pred_lt`?

**Review by YaelDillies — COMMENTED** [review]
Makes sense. Now that you have updated the content of the PR, can you update its title and description?

**Comment by YaelDillies** [comment]
Thanks!

maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

File: Mathlib/Order/Interval/Set/Monotone.lean [hunk 3]
```diff
@@ -188,10 +188,18 @@ theorem strictMonoOn_Iic_of_lt_succ [SuccOrder α] [IsSuccArchimedean α] {n : 
...
+    (hψ : ∀ m, ψ m < ψ (succ m)) : StrictMono ψ := fun _ _ h ↦
+  (strictMonoOn_Iic_of_lt_succ fun m _ ↦ hψ m) h.le le_rfl h
+
 theorem strictAntiOn_Iic_of_succ_lt [SuccOrder α] [IsSuccArchimedean α] {n : α}
     (hψ : ∀ m, m < n → ψ (succ m) < ψ m) : StrictAntiOn ψ (Set.Iic n) := fun i hi j hj hij =>
   @strictMonoOn_Iic_of_lt_succ α βᵒᵈ _ _ ψ _ _ n hψ i hi j hj hij
 
+theorem strictAnti_of_pred_lt [SuccOrder α] [IsSuccArchimedean α]
+    (hψ : ∀ m, ψ (succ m) < ψ m) : StrictAnti ψ := fun _ _ h ↦
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/Interval/Set/Monotone.lean:200]: Should this not be ...succ_lt?

Comment (vihdzp) [reply Mathlib/Order/Interval/Set/Monotone.lean:200]: Yeah, and also, we should have `pred` versions of these statements too.

Comment (YaelDillies) [reply Mathlib/Order/Interval/Set/Monotone.lean:200]: Wow sorry, I was clearly tired when I maintainer merged this. @YnirPaz, mind opening a follow-up PR?

Comment (YnirPaz) [reply Mathlib/Order/Interval/Set/Monotone.lean:200]: Wow sorry, these names are somehow short-circuiting my brain. #16813

**Pull request closed without merging**